app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -10,6 +11,9 @@ def resource_locks
 
       def plan(job_invocation, host, template_invocation, proxy, connection_options = {})
         action_subject(host, :job_name => job_invocation.job_name)
+        link!(job_invocation)
+        link!(template_invocation)
+
```

Comment:
The reason for moving those three lines is to create the links before any error can be raised?